### DISCUSSION TITLE
Failing test (CRASH)

### DISCUSSION THREAD

**Discussion by pastoreg** [idaholab/moose#17361] [Q&A Modules: Solid mechanics] [answered]
Hi all,

I have an MR that is failing because of a test failure (CRASH).  It seems the test runs fine in serial or with mpiexec, however, ./run_tests --n-threads=2 reproduces the error, as below.  I am not sure what may be causing this and would appreciate any help.  Thanks!

(moose) gpastore@x86_64-apple-darwin13 bison % ./run_tests --n-threads=2 --re=test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm: Exit Code: -11

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm: ################################################################################

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm: Tester failed, reason: CRASH

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm: 

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm [...] FAILED (CRASH)

Final Test Results:

--------------------------------------------------------------------------------------------------------------

test:tensor_mechanics/fecral_plasticity/power-law_strain_hardening.test_power-law_plasticity_fecral_tm [...] FAILED (CRASH)

--------------------------------------------------------------------------------------------------------------

Ran 1 tests in 1.5 seconds. Average test time 0.2 seconds, maximum test time 0.2 seconds.

Giovanni

--
Giovanni Pastore, PhD
Research Associate Professor
Department of Nuclear Engineering
University of Tennessee
Knoxville, TN 37916
gpastore@utk.edu

**Comment by pastoreg** [comment, marked as answer]
Never mind, I figured this out.

The crash was due to a segfault caused by a pretty simple error.  Below is a code snippet from a function called by my constructor:

if (!_fe_problem.hasFunction("FeCrAl_hardening_function_296K"))
{
    const std::vector<Real> hf0_plastic_strains = {0.0e+00,
                                                   1.0e-03,
                                                   2.0e-03,
                                                   3.0e-03,
                                                   4.0e-03,
                                                   5.0e-03,
                                                   1.0e-02,
                                                   2.0e-02,
                                                   3.0e-02,
                                                   4.0e-02,
                                                   5.0e-02,
                                                   6.0e-02,
                                                   7.0e-02,
                                                   8.0e-02,
                                                   9.0e-02,
                                                   1.0e-01,
                                                   1.1e-01,
                                                   1.6e-01};
    const std::vector<Real> hf0_stresses = {4.55e+08,
                                            4.65e+08,
                                            4.71e+08,
                                            4.75e+08,
                                            4.79e+08,
                                            4.82e+08,
                                            4.94e+08,
                                            5.12e+08,
                                            5.26e+08,
                                            5.39e+08,
                                            5.51e+08,
                                            5.63e+08,
                                            5.74e+08,
                                            5.85e+08,
                                            5.96e+08,
                                            6.07e+08,
                                            6.17e+08,
                                            6.72e+08};

    InputParameters hf0_params = _app.getFactory().getValidParams("PiecewiseLinear");
    hf0_params.set<std::vector<Real>>("x") = hf0_plastic_strains;
    hf0_params.set<std::vector<Real>>("y") = hf0_stresses;
    _fe_problem.addFunction("PiecewiseLinear", "FeCrAl_hardening_function_296K", hf0_params);
}
  _hf_temperatures.push_back(296.);
  _hardening_functions.push_back(dynamic_cast<const PiecewiseLinear *>(
      &_fe_problem.getFunction("FeCrAl_hardening_function_296K")));

In the version that was giving the test crash (segfaulting), the last two lines were inside the if statement, hence, the _hf_temperatures and _hardening_functions ended up having size zero in multi-thread.

Thanks,
Giovanni

**Reply by aeslaughter** [reply]
Thanks for posting the solution.
